modules/private.php

```diff
@@ -0,0 +1,256 @@
+         </div>
+		<?php
+	}
+
+	/**
+	 * Don't let search engines index private sites
+	 * or sites not deemed publicly available, like deleted, archived, spam.
+	 *
+	 * @param string $output Robots.txt output.
+	 */
+	static function private_robots_txt( $output ) {
+		$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+		$output .= "Disallow: /\n";
+		return $output;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	static function private_blog_ajax_nonce_check( $action, $result ) {
+		global $current_user, $wpdb;
+
+		if ( is_super_admin() ) {
+			return;
+		}
+
+		if ( 1 !== $result && 2 !== $result ) {
+			return;
+		}
+
+		if ( 'find-posts' !== $action && 'internal-linking' !== $action ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {
+			return;
+		}
+
+		if ( ! Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) {
+			wp_die( -1 );
+		}
+	}
+
+	/**
+	 * Disables WordPress Rest API for external requests
+	 */
+	static function disable_rest_api() {
+		global $pagenow, $current_user, $wpdb;
+
+		if ( $current_user && ( is_super_admin() || Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) ) {
+			return;
+		}
+
+		die( __( 'This site is private.', 'jetpack' ) );
+	}
+
+	static function module_override( $modules ) {
+		$disabled_modules = array(
+			'publicize',
+			'sharedaddy',
+			'subscriptions',
+			'json-api',
+			'enhanced-distribution',
+			'google-analytics',
+			'photon',
+			'sitemaps',
+			'verification-tools',
+			'wordads',
+		);
+
+		foreach ( $disabled_modules as $module_slug ) {
+			$found = array_search( $module_slug, $modules );
+			if ( false !== $found ) {
+				unset( $modules[ $found ] );
+			}
+		}
+
+		return $modules;
+	}
+
+	/**
+	 * Show an error when the blog_public option is updated
+	 */
+	static function private_update_option_blog_public() {
+		if ( function_exists( 'add_settings_error') ) {
+			add_settings_error( 'general', 'setting_not_updated', __( "This setting is ignored because you made your site private in Jetpack", 'jetpack' ), 'error' );
+		}
+	}
+
+	/**
+	 * Adds a message to the 'At a Glance' dashboard widget.
+	 */
+	static function add_private_dashboard_glance_items( $content ) {
+		return $content . '<br><br>' . __( 'This site is currently Private' );
```

Comment:
Should we add a link to the Jetpack settings here as well? I am also wondering if "Private" should be capitalized?

Reply:
I don't think private needs to be capitalised, as it should be a familiar term. It would make more sense if it was a specific feature name.